video/out/vulkan/context_display.c

```diff
@@ -51,24 +51,32 @@ struct mode_selector {
 static bool walk_display_properties(struct mp_log *log,
                                     int msgl_err,
                                     VkPhysicalDevice device,
+                                    VkInstance inst,
                                     struct mode_selector *selector) {
     bool ret = false;
     VkResult res;
 
     int msgl_info = selector ? MSGL_TRACE : MSGL_INFO;
 
+    // Load all vulkan functions.
+    VK_LOAD_FUN(inst, GetPhysicalDeviceProperties);
+    VK_LOAD_FUN(inst, GetPhysicalDeviceDisplayPropertiesKHR);
+    VK_LOAD_FUN(inst, GetPhysicalDeviceDisplayPlanePropertiesKHR);
+    VK_LOAD_FUN(inst, GetDisplayPlaneSupportedDisplaysKHR);
+    VK_LOAD_FUN(inst, GetDisplayModePropertiesKHR);
```

Comment:
For vkGetInstanceProcAddr?

Reply:
No, for all the function pointers you get via calls to it.

Reply:
Ah duh, good point.

Reply:
@rcombs vulkan spec guarantees that if one of the functions from an extension is available, all of them will be - so one check for the display-related ones will suffice. (And `vkGetPhysicalDeviceProperties` is required to exist also by spec)

YMMV on whether or not we're allowed to segfault if the vulkan driver violates spec, but I think the answer is yes - otherwise you would not be able to rely on any function working

Reply:
Yeah, but there has to be _a_ check here, since this can run without anything else from the extension having been checked previously.